[docs] Add documentation for the column menu (#815)
mui-org/material-ui-x: docs/pages/api-docs/data-grid.md

| <span class="prop-name">columnTypes</span> | <span class="prop-type">ColumnTypesRecord</span> |   | Extend native column types with your new column types. |
| <span class="prop-name">components</span> | <span class="prop-type">GridComponentOverridesProp</span> |   | Overrideable components. |
| <span class="prop-name">density</span> | <span class="prop-type">Density</span> | standard | Sets the density of the grid. |

> | <span class="prop-name">disableColumnMenu</span> | <span class="prop-type">boolean</span> | false | If `true`, the column menu will be disabled. |

Reviewer: 1. We use the present tense
```suggestion
| <span class="prop-name">disableColumnMenu</span> | <span class="prop-type">boolean</span> | false | If `true`, the column menu is disabled. |
```

2. We need to update the two other descriptions of the prop `disableColumnMenu?: boolean;` to match.

Reviewer: 1. is point number 4 of https://trello.com/c/1Q7hqXHu/2714-technical-documentation-guidelines.

Reviewer: What about point 2?

Author: There is only the `x-grid.md` file where the prop is mentioned, just updated it.

Reviewer: > We need to update the two other descriptions of the prop

I meant in the codebase, not in the documentation

Author: One bonus place, inside the `colDef.ts` there is another prop called `disableColumnMenu` that is specific for that column. I've updated all of them.
